src/SignalR/server/StackExchangeRedis/src/Internal/RedisProtocol.cs

```diff
@@ -43,30 +43,31 @@ public byte[] WriteInvocation(string methodName, object[] args, IReadOnlyList<st
             // * [The output of WriteSerializedHubMessage, which is an 'arr']
             // Any additional items are discarded.
 
-            var writer = MemoryBufferWriter.Get();
+            var memoryBufferWriter = MemoryBufferWriter.Get();
+            var writer = new MessagePackWriter(memoryBufferWriter);
```

Comment:
nit: move this into the `try` statement

Reply:
`var writer = MemoryBufferWriter.Get();` cannot be moved inside the `try` block as the variable is required in `finally` block in order to do the `.Return()` to the pool.